src/SFA.DAS.PSRService.Web/Startup.cs

```diff
@@ -98,14 +105,103 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
         {
             if (env.IsDevelopment())
             {
-                app.UseBrowserLink();
+                //app.UseBrowserLink();
                 app.UseDeveloperExceptionPage();
             }
             else
             {
                 app.UseExceptionHandler("/Home/Error");
             }
 
+            // Add Content Security Policy           
+            app.UseCsp(options => options
+                .DefaultSources(s =>
+                {
+                    s.Self()
+                        .CustomSources(
+                        "https://*.zdassets.com",
+                        "https://*.zendesk.com",
+                        "wss://*.zendesk.com",
+                        "wss://*.zopim.com",
+                        "https://embed-euw1.rcrsv.io/",
+                        "https://assets.publishing.service.gov.uk"
+                        );
+                })
+                .StyleSources(s =>
+                {
+                    s.Self()
+                    .CustomSources("https://www.googletagmanager.com/",
+                                    "https://www.tagmanager.google.com/",
+                                    "https://tagmanager.google.com/",
+                                    "https://fonts.googleapis.com/",
+                                    "https://*.zdassets.com",
+                                    "https://*.zendesk.com",
+                                    "wss://*.zendesk.com",
+                                    "wss://*.zopim.com"
+                                    );
+
+                    //Google tag manager uses inline styles when administering tags. This is done on PREPROD only
+                    //TinyMCE uses inline styles
+                    s.UnsafeInline();
+                }
+                )
+                .ScriptSources(s =>
+                {
+                    s.Self()
+                        .CustomSources("https://az416426.vo.msecnd.net/scripts/a/ai.0.js",
+                                    "*.google-analytics.com",
+                                     "*.googleapis.com",
+                                    "*.googletagmanager.com/",
+                                   "https://www.tagmanager.google.com/",
+                                    "https://*.zdassets.com",
+                                    "https://*.zendesk.com",
+                                    "wss://*.zendesk.com",
+                                    "wss://*.zopim.com",
+                                    "https://embed-euw1.rcrsv.io");
+                    //Google tag manager uses inline scripts when administering tags. This is done on PREPROD only
+                    if (env.IsEnvironment(EnvironmentNames.PREPROD))
```

Comment:
I see what that's for now - but checkout the similar comment above which is not wrapped in a env.IsEnvironment if block

Reply:
You've changed that to IsStaging - but what does that do?, also see above where there is a similar comment about only doing this in PREPROD which isn't guarded.